applications/dds_video/README.md

```diff
@@ -0,0 +1,96 @@
+# DDS Video Application
+
+The DDS Video application demonstrates how video frames can be written to or
+read from a DDS databus in order to provide flexible integration between
+Holoscan applications and other applications (using Holoscan or not) via DDS.
+
+The application can be run as either a publisher or as a subscriber. In either case,
+it will use the [VideoFrame](../../operators/dds/video/VideoFrame.idl) data topic
+registered by the `DDSVideoPublisherOp` or `DDSVideoSubscriberOp` operators in order
+to write or read the video frame data to/from the DDS databus, respectively.
+
+When run as a publisher, the source for the input video frames will come from an
+attached V4L2-compatible camera via the `V4L2VideoCaptureOp` operator.
+
+When run as a subscriber, the application will use Holoviz to render the received
+video frames to the display. In addition to the video stream, the subscriber
+application will also subscribe to the `Square`, `Circle`, and `Triangle` topics
+as used by the [RTI Shapes Demo](https://www.rti.com/free-trial/shapes-demo).
+Any shapes received by this subscriber will also be overlaid on top of the
+Holoviz output.
+
+![DDS Video Application Workflow](docs/workflow_dds_video_app.png)
+
+## Building the Application
+
+This application requires [RTI Connext](https://content.rti.com/l/983311/2024-04-30/pz1wms)
+be installed and configured with a valid RTI Connext license prior to use.
+
+To build the application, the `RTI_CONNEXT_DDS_DIR` CMake variable must point to
+the installation path for RTI Connext. This can be done automatically by setting
+the `NDDSHOME` environment variable to the RTI Connext installation directory
+(such as when using the RTI `setenv` scripts), or manually at build time, e.g.:
+
+```sh
+$ ./run build dds_video --configure-args -DRTI_CONNEXT_DDS_DIR=~/rti/rti_connext_dds-6.1.2
+```
+
+## Running the Application
+
+Both a publisher and subscriber process must be launched to see the result of
+writing to and reading the video stream from DDS, respectively.
+
+To run the publisher process, use the `-p` option:
```

Comment:
Notes from attempting to build and run in a HoloHub dev container:

- RTI Connext installation seems to complete, but launcher fails due to missing packages:
```
irst time running RTI Connext DDS... 
Copying examples into /home/holoscan/rti_workspace/7.3.0/examples...
tbirdsong@c6619bc-lcedt:/workspace/holohub$ Copying user configuration files into /home/holoscan/rti_workspace/7.3.0/user_config...
Exception in thread "main" java.lang.UnsatisfiedLinkError: Could not load SWT library. Reasons: 
	no swt-gtk-4963r5 in java.library.path: /home/holoscan/rti_connext_dds-7.3.0/resource/app/jre/x64Linux/lib/server:/home/holoscan/rti_connext_dds-7.3.0/resource/app/jre/x64Linux/lib:/home/holoscan/rti_connext_dds-7.3.0/resource/app/jre/x64Linux/../lib:/home/holoscan/rti_connext_dds-7.3.0/bin/../resource/app/lib/x64Linux3gcc4.8.2/:/home/holoscan/rti_connext_dds-7.3.0/bin/../resource/app/lib/x64Linux3gcc4.8.2:/home/holoscan/rti_connext_dds-7.3.0/bin/../resource/app/jre/x64Linux/lib/server::/usr/local/cuda/compat/lib:/usr/local/nvidia/lib:/usr/local/nvidia/lib64:/opt/nvidia/holoscan/lib:/usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib
	no swt-gtk in java.library.path: /home/holoscan/rti_connext_dds-7.3.0/resource/app/jre/x64Linux/lib/server:/home/holoscan/rti_connext_dds-7.3.0/resource/app/jre/x64Linux/lib:/home/holoscan/rti_connext_dds-7.3.0/resource/app/jre/x64Linux/../lib:/home/holoscan/rti_connext_dds-7.3.0/bin/../resource/app/lib/x64Linux3gcc4.8.2/:/home/holoscan/rti_connext_dds-7.3.0/bin/../resource/app/lib/x64Linux3gcc4.8.2:/home/holoscan/rti_connext_dds-7.3.0/bin/../resource/app/jre/x64Linux/lib/server::/usr/local/cuda/compat/lib:/usr/local/nvidia/lib:/usr/local/nvidia/lib64:/opt/nvidia/holoscan/lib:/usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib
	no swt in java.library.path: /home/holoscan/rti_connext_dds-7.3.0/resource/app/jre/x64Linux/lib/server:/home/holoscan/rti_connext_dds-7.3.0/resource/app/jre/x64Linux/lib:/home/holoscan/rti_connext_dds-7.3.0/resource/app/jre/x64Linux/../lib:/home/holoscan/rti_connext_dds-7.3.0/bin/../resource/app/lib/x64Linux3gcc4.8.2/:/home/holoscan/rti_connext_dds-7.3.0/bin/../resource/app/lib/x64Linux3gcc4.8.2:/home/holoscan/rti_connext_dds-7.3.0/bin/../resource/app/jre/x64Linux/lib/server::/usr/local/cuda/compat/lib:/usr/local/nvidia/lib:/usr/local/nvidia/lib64:/opt/nvidia/holoscan/lib:/usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib
	Can't load library: /home/tbirdsong/libswt-gtk-4963r5-64.so
	Can't load library: /home/tbirdsong/libswt-gtk-64.so
	Can't load library: /home/tbirdsong/libswt-64.so

	at org.eclipse.swt.internal.Library.loadLibrary(Library.java:345)
	at org.eclipse.swt.internal.Library.loadLibrary(Library.java:254)
	at org.eclipse.swt.internal.C.<clinit>(C.java:19)
	at org.eclipse.swt.internal.Converter.wcsToMbcs(Converter.java:209)
	at org.eclipse.swt.internal.Converter.wcsToMbcs(Converter.java:155)
	at org.eclipse.swt.widgets.Display.<clinit>(Display.java:169)
	at com.rti.tools.launcher.main.LauncherApp.<clinit>(LauncherApp.java:127)
```
- Build command above succeeds without issue
- Publisher fails to launch, possibly because I could not open the launcher to specify a license file?
```
workspace/holohub/build/applications/dds_video$ ./dds_video -p
[info] [dds_video.cpp:247] Starting publisher for stream 0 in domain 0
[info] [gxf_executor.cpp:247] Creating context
[info] [gxf_executor.cpp:1672] Loading extensions from configs...
RTI Connext DDS No source for License information
Please contact support@rti.com with any questions or comments.
[error] [gxf_executor.cpp:1369] Exception occurred during initialization of operator: 'dds' - Failed to create DomainParticipant
terminate called after throwing an instance of 'dds::core::Error'
  what():  Failed to create DomainParticipant
Aborted (core dumped)
```

Reply:
Resolved:
- App does not support HoloHub container (bare metal only)
- `NDDSHOME` must be set for license to be found

Could you please update the README with those points?

Reply:
Added notes to README